Functional and local options

uhop opened the pull request:
As per https://github.com/koajs/compress/issues/121#issuecomment-653988740

Includes new tests, passes ESLint, and the existing tests.

3imed-jaberi reviewed (commented):
LGTM, just should update as #135

uhop commented:
@jonathanong There is a non-responding `build (16.x)` job. I don't know how to kill it or re-run it.

I refreshed the PR. The principal code is unchanged, but now it contains face lifts:

- I copied and replaced the harness from `koajs/koa` to replace dead dependencies.
   - No need for numerous version bumps from `dependabot`.
- I updated Node versions in GitHub actions to include only supported versions (https://endoflife.date/nodejs).
   - Bumped version of obsolete actions.
   - Updated syntax.
- Minor style changes to accommodate `standard`.

I propose to use this PR as a foundation so I can address other technical issues, specifically:

- https://github.com/koajs/compress/issues/250
- https://github.com/koajs/compress/issues/251

That would be a good point to tag a new release.

The rest of issues can be discussed and resolved after that.

jonathanong commented:
@uhop fixed, just needed to update settings
